src/unix_connection.rs

```diff
@@ -12,6 +12,7 @@ impl<IoHandler> UnixConnection<IoHandler>
 where
     IoHandler: io::Read + io::Write + Send + Sync,
 {
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
Yeah, lots of fun with these... I have mostly been adding an `allow` for these as well, although sometimes there are better solutions.

Reply:
I guess this could be a `impl From<IoHandler> for UnixConnection<IoHandler>` instead of `new`. On the other hand we do not err in any case and could also change the method below to:
```suggestion
/// Create a new `UnixConnection` for the given socket
pub fn new(socket: IoHandler) -> Self {
    Self { socket }
}
```